data/awsrds/v0/main.go

```diff
@@ -0,0 +1,448 @@
+		}
+	}
+
+	return nil, fmt.Errorf("error creating SQL client")
+}
+
+// newClientIAM creates a new SQL client with IAM authentication
+func newClientIAM(e *execution) (*sqlx.DB, error) {
+	cfg := LoadConfig(e)
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(cfg.Region),
+		Credentials: credentials.NewStaticCredentials(
+			cfg.DBAWSAccessKeyID, cfg.DBAWSSecretAccessKey, ""),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	DBEndpoint := fmt.Sprintf("%s:%s", cfg.DBHost, cfg.DBPort)
+
+	authToken, err := rdsutils.BuildAuthToken(
+		DBEndpoint,
+		cfg.Region,
+		cfg.DBUser,
+		sess.Config.Credentials,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Test every engines to find the correct one
+	var db *sqlx.DB
+	for _, engine := range engines {
+		dsn := fmt.Sprintf(engine+"?tls=true&tls=skip-verify&allowCleartextPasswords=true",
+			cfg.DBUser, authToken, DBEndpoint, cfg.DBName,
+		)
+
+		db, err = sqlx.Open("mysql", dsn)
+		if err == nil {
+			return db, nil
+		}
+	}
+
+	return nil, fmt.Errorf("error creating SQL client")
+}
+
+func getUser(setup *structpb.Struct) string {
+	return setup.GetFields()["user"].GetStringValue()
+}
+func getPassword(setup *structpb.Struct) string {
+	return setup.GetFields()["password"].GetStringValue()
+}
+func getName(setup *structpb.Struct) string {
+	return setup.GetFields()["name"].GetStringValue()
+}
+func getHost(setup *structpb.Struct) string {
+	return setup.GetFields()["host"].GetStringValue()
+}
+func getPort(setup *structpb.Struct) string {
+	port := setup.GetFields()["port"].GetNumberValue()
+	portStr := strconv.FormatFloat(port, 'f', -1, 64)
+	return portStr
+}
+func getRegion(setup *structpb.Struct) string {
+	return setup.GetFields()["region"].GetStringValue()
+}
+func getAWSAccessKeyID(setup *structpb.Struct) string {
+	return setup.GetFields()["aws-access-key-id"].GetStringValue()
+}
+func getAWSSecretAccessKey(setup *structpb.Struct) string {
+	return setup.GetFields()["aws-secret-access-key"].GetStringValue()
+}
+
+func (e *execution) Execute(ctx context.Context, inputs []*structpb.Struct) ([]*structpb.Struct, error) {
+	outputs := []*structpb.Struct{}
+
+	var client *sqlx.DB
+	var err error
+
+	// check if the password field is present, if it is, create a client with username and password else create a client with IAM
+	if e.Setup.GetFields()["password"] != nil {
+		client, err = newClient(e)
+	} else if e.Setup.GetFields()["password"] == nil && e.Setup.GetFields()["aws-access-key-id"] != nil && e.Setup.GetFields()["aws-secret-access-key"] != nil {
+		client, err = newClientIAM(e)
+	} else {
+		return nil, fmt.Errorf("some fields are missing, please fill in the required fields")
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("error creating SQL client: %v", err)
+	}
+	defer client.Close()
+
+	for _, input := range inputs {
+		var output *structpb.Struct
+
+		switch e.Task {
+		case taskUpsert:
+			tableName := input.GetFields()["table-name"].GetStringValue()
```

Comment:
Important
I think making dependency clean is important.

Could you take a look [this comment](https://github.com/instill-ai/component/pull/187#discussion_r1662946437)?

TLDR
I want to make protobuf struct to component struct.
In our other services, we depend on component struct rather than protobuf struct.